src/estimators/exact_set.py

```diff
@@ -67,11 +77,18 @@ def __init__(self):
     EstimatorBase.__init__(self)
 
   def __call__(self, sketch_list):
-    """Return len(sketch)."""
-    union = ExactSet()
+    """Returns sketch frequency histogram."""
+    if len(sketch_list) == 0:
+      return [0]
+    union = ExactMultiSet()
     for s in sketch_list:
-      union.add_ids(s.ids())
-    return [len(union)]
+      for id in s.ids():
+        union.add_ids([id] * s.frequency(id))
+    frequency_counts = collections.defaultdict(int)
+    for x in union.ids():
+      frequency_counts[union.frequency(x)] += 1
+    histogram = list(reversed(np.cumsum([frequency_counts[i] for i in range(max(frequency_counts), 0, -1)])))
```

Comment:
This is a little much to have in a single list comprehension. :)

Reply:
I've added some comments explaining what is happening.

Reply:
The list comprehension is too long/complex. A couple of temporary variables would make the comment moot too I think.